src/sentry/migrations/0375_create_notificationaction.py

```diff
@@ -0,0 +1,76 @@
+import sentry.db.models.fields.foreignkey
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0374_breaking_even_more_keys"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="NotificationAction",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("type", models.SmallIntegerField()),
+                ("target_type", models.SmallIntegerField()),
+                ("target_identifier", models.TextField(null=True)),
+                ("target_display", models.TextField(null=True)),
+                ("trigger_type", models.SmallIntegerField()),
+                (
+                    "integration",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        null=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        to="sentry.Integration",
+                    ),
+                ),
+                (
+                    "organization",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to="sentry.Organization"
+                    ),
+                ),
+                (
+                    "project",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        null=True, on_delete=django.db.models.deletion.CASCADE, to="sentry.Project"
+                    ),
+                ),
+                (
+                    "sentry_app",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        null=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        to="sentry.SentryApp",
+                    ),
+                ),
```

Comment:
This implies that customers will have to repeat the integration configuration for each project they want to enable notifications for. Should we have a 1:N with projects instead?

```mermaid
erDiagram

Organization ||--|{ NotificationAction : "has many"
NotificationAction ||--o| Integration : "maybe"
NotificationAction ||--o| SentryApp : "maybe"
NotificationAction ||--|{ Project : "one or more"
```

Something like this would allow each notification 'action' to be shared across multiple projects or no projects. Having no project association would be useful for other billing notifications.

Reply:
That makes sense, I'll use a ManyToManyField instead